File: Beecrowd/Beginner/1435 - 1985/1914 - Whose Turn Is It/Python/1914_WhoseTurnIsIt.py
testCases = int(input())

for _ in range(testCases):
    player1, choice1, player2, choice2 = input().split()
    number1, number2 = map(int, input().split())

    if (number1 + number2) % 2 == 0:
        if choice1 == "PAR":
            print(player1)
        else:
            print(player2)
    else:
        if choice1 == "IMPAR":
            print(player1)
        else:
            print(player2)

File: Beecrowd/Beginner/1435 - 1985/1914 - Whose Turn Is It/C++/1914_WhoseTurnIsIt.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int testCases;
    cin >> testCases;

    while (testCases--) {
        string player1, player2, choice1, choice2;
        int number1, number2;
        cin >> player1 >> choice1 >> player2 >> choice2 >> number1 >> number2;

        if ((number1 + number2) % 2 == 0) {
            if (choice1 == "PAR") {
                cout << player1 << endl;
            } else cout << player2 << endl;
        } else {
            if (choice1 == "IMPAR") {
                cout << player1 << endl;
            } else cout << player2 << endl;
        }
    }

    return 0;
}